### DISCUSSION TITLE
Rest API and n:m relations

### DISCUSSION THREAD

**Discussion by 321dev** [directus/directus#8940] [Archive]
Hi all.

I'm experimenting with n:m relations, i think i have the table(s) on the Directus side set up correctly, but i'm struggling with the Rest API output.

[...]/items/test1?fields=* => gives this output:
```
{
  "data": [{
      "id": 1,
      "text": "gggg hhhh",
      "types": [3, 4]
    }, {
      "id": 2,
      "text": "aaa bbb",
      "types": [5]
    }
  ]
}

```

items/test1?fields=*,types.typelist_id.type => gives this output:
```
{
  "data": [{
      "id": 1,
      "text": "gggg hhhh",
      "types": [{
          "typelist_id": {
            "type": "morgen"
          }
        }, {
          "typelist_id": {
            "type": "tisch"
          }
        }
      ]
    }, {
      "id": 2,
      "text": "aaa bbb",
      "types": [{
          "typelist_id": {
            "type": "tisch"
          }
        }
      ]
    }
  ]
}

```

I'd like to get only the final leaf of the relational part, the content of the "type" field.
Like this:
```
{
  "data": [{
      "id": 1,
      "text": "gggg hhhh",
      "types": ["morgen", "tisch"]
    }, {
      "id": 2,
      "text": "aaa bbb",
      "types": ["tisch"]
    }
  ]
}
```

Is that possible?

**Comment by joselcvarela** [comment]
Hello @321dev,
What are you trying to achieve is a result from an aggregate function (for example, `array_agg` on postgres)
The only supported aggregators right now are defined here: https://docs.directus.io/reference/api/query/#aggregation-grouping

All in all, that is not possible.

**Reply by 321dev** [reply]
Thanks for your quick answer. Now i know that i can stop trying.
